torch/_higher_order_ops/templated_attention.py

```diff
@@ -218,3 +248,437 @@ def templated_attention_fake_tensor_mode(
             batch_size, num_heads, seq_len_q, dtype=torch.float32
         )
         return torch.empty_like(query, memory_format=torch.contiguous_format), logsumexp
+
+
+# ---------------------------- Autograd Implementation ----------------------------
+# # TODO We need to implement an autograd function for this, there is some complexity to do this generically
+# templated_attention.py_impl(DispatchKey.Autograd)(
+#     autograd_not_implemented(templated_attention, deferred_error=True)
+# )
+
+from torch._dispatch.python import suspend_functionalization
+from torch._functorch.aot_autograd import AOTConfig, create_joint, from_fun
+
+from torch._subclasses.functional_tensor import (
+    disable_functional_mode,
+    FunctionalTensor,
+)
+from torch.fx.experimental.proxy_tensor import disable_proxy_modes_tracing
+
+# from torch._higher_order_ops.utils import _unstack_pytree, _stack_pytree
+
+
+dummy_aot_config = AOTConfig(
+    fw_compiler=None,  # type: ignore[arg-type]
+    bw_compiler=None,  # type: ignore[arg-type]
+    partition_fn=None,  # type: ignore[arg-type]
+    decompositions={},
+    num_params_buffers=0,
+    aot_id=0,
+    keep_inference_input_mutations=False,
+)
+
+
+def create_fw_bw_graph(
+    score_mod: Union[Callable, UserFunctionVariable], index_values, other_buffers
+):
```

Comment:
Some thought will need to be put in to make this more generic between templated_attn and map.... hmm

Reply:
Also do to circular imports the import would need to be moved within the autograd func for flexattention